src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -68,6 +73,29 @@ public RecordService()
   {
   }
 
+  public void setMetadataConfig(Map<String, String> metadataConfigIn) {
+    metadataConfig = metadataConfigIn;
+  }
+
+  /**
+   * Check whether the input key is in the metadata configuration.
+   * If not, we keep the metadata, which preserves previous behaviour.
+   * If the key is in the config and not equal to the default value,
+   * we will remove the metadata.
+   * @param key a String indicating the key to check in the config map
+   * @return whether to keep the metadata
+   */
+  private boolean checkMetaFlag(String key)
+  {
+    if (!metadataConfig.containsKey(key))
+    {
+      return true;
+    }
+    else
+    {
+      return metadataConfig.get(key).equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT);
+    }
+  }
```

Comment:
Same thing, if we create a Metadata Config object, we can check it in SinkTask, and we don't need check it here.

Reply:
will change that